src/lib/rules/apple-touch-icons/apple-touch-icons.ts

```diff
@@ -0,0 +1,268 @@
+        };
+
+        const chooseBestIcon = (icons: Array<IAsyncHTMLElement>): IAsyncHTMLElement => {
+
+            // Site will usually have something such as:
+            //
+            // <link rel="apple-touch-icon" sizes="60x60" href="/apple-touch-icon-60x60.png">
+            // <link rel="apple-touch-icon" sizes="72x72" href="/apple-touch-icon-72x72.png">
+            // <link rel="apple-touch-icon" sizes="76x76" href="/apple-touch-icon-76x76.png">
+            // <link rel="apple-touch-icon" sizes="114x114" href="/apple-touch-icon-114x114.png">
+            // <link rel="apple-touch-icon" sizes="120x120" href="/apple-touch-icon-120x120.png">
+            // <link rel="apple-touch-icon" sizes="144x144" href="/apple-touch-icon-144x144.png">
+            // <link rel="apple-touch-icon" sizes="152x152" href="/apple-touch-icon-152x152.png">
+            // <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon-180x180.png">
+            // <link rel="apple-touch-icon" href="/apple-touch-icon-57x57.png">
+            //
+            // so what this function will try to do is select the
+            // icon that will most likely generate the fewest errors.
+
+            let bestIcon;
+
+            for (const icon of icons) {
+                const sizes = normalizeString(icon.getAttribute('sizes'));
+
+                if (!sizes) {
+                    return icon;
+                } else if (sizes === '180x180') {
+                    bestIcon = icon;
+                }
+            }
+
+            return bestIcon ? bestIcon : icons[0];
+        };
+
+
+        const validate = async (event: ITraverseEnd) => {
+            const { resource }: { resource: string } = event;
+
+            // The following checks don't make sense for non-HTML documents.
+
+            if (!isHTMLDocument(resource, context.pageHeaders)) {
+                return;
+            }
+
+            // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+            const pageDOM: IAsyncHTMLDocument = context.pageDOM as IAsyncHTMLDocument;
+            const appleTouchIcons: Array<IAsyncHTMLElement> = getAppleTouchIcons(await pageDOM.querySelectorAll('link'));
+
+            if (appleTouchIcons.length === 0) {
+                await context.report(resource, null, `No 'apple-touch-icon' was specified`);
+
+                return;
+            }
+
+            // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+            // Choose the icon that will most likely
+            // pass most of the following tests.
+
+            const appleTouchIcon: IAsyncHTMLElement = chooseBestIcon(appleTouchIcons);
+
+            // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+            // Check if `rel='apple-touch-icon'`.
+            // See `getAppleTouchIcons` function for more details.
+
+            if (normalizeString(appleTouchIcon.getAttribute('rel')) !== 'apple-touch-icon') {
```

Comment:
It looks like in `getAppleTouchIcons` you're also checking if `normalizeString(appleTouchIcon.getAttribute('rel'))` includes `apple-touch-icon-precomposed`. Is it necessary to do that here too?

Reply:
>  Is it necessary to do that here too?

Yes since you can have something such as: `REL="APPLE-TOUCH-ICON"`